docs/pai-management/doc/add-service.md

```diff
@@ -46,79 +58,47 @@ If your service image could be pulled from a public docker registry, you could s
-    - Command: ```cp -r pai/pai-management/$src pai/pai-management/$dst ```. ```src``` and ```dst``` is the value in this part.
 
 #### Place the Image Directory into PAI <a name="Image_Place"></a>
 
- Note that the name of image's directory should be same with the image name.
-
-For example, now we wanna add a docker image "XYZ" into pai. You will first create a directory named "XYZ" in the path ```pai/pai-management/src/XYZ```. That is the image's directory named as the image's name.
-
-
-If you wanna paictl to build hbase image, you should move the director ```example/src/hbase``` to ```pai/pai-management/src```.
+- At first, according to the service name, you should create a directory under the path [pai/src/](../../../src). In this example, a directory named hbase will be created.
+- Secondly, a directory named ```build``` should be created under "src/hbase/"
+- At last, just put all the source code from the path [example/add-service/hbase/build](example/add-service/hbase/build), and then paste them into the the path ```src/hbase/build```  
 
 
-#### Place the Image Directory into PAI <a name="Image_Build_Push"></a>
+#### Build and push image with pai-build <a name="Image_Build_Push"></a>
 
 ```
-./paictl.py image build -p /path/to/your/cluster-configuration/dir -n hbase
-
-
-./paictl.py image push -p /path/to/your/cluster-configuration/dir -n hbase
-
+cd build/
+./pai_build.py build -c /path/to/configuration-dir/  -s hbase
+./pai_build.py push -c /path/to/configuration-dir/ -i hbase
 ```
 
 
-## Add Service's Bootstrap Module <a name="Boot"></a>
+## Add Service's Deployment Module <a name="Boot"></a>
 
-After hbase image is built, you need bootstrap it in pai. Now the service management system is kubernetes.
+After hbase image is built, you need bootstrap it in openPai. Now the service management system is kubernetes.
 
 
 #### Prepare Service Configuration <a name="service_configuration"></a>
 
-This is the configuration of your service bootstrap module. And paictl will call different script to handle different things. This file should be placed in your service bootstrap module. And its name should be ```service.yaml```
+This is the configuration of your service deployment module. And paictl will call different script to handle different things. This file should be placed in your service deployment module. And its name should be ```service.yaml```
 
-[Hbase's bootstrap module's configuration](example/add-service/bootstrap/hbase/service.yaml)
+[Hbase's bootstrap module's configuration](example/add-service/hbase/deploy/service.yaml)
 
 Here is the service configuration of HBase.
 
 ```YAML
 # Tell paictl which service should be ready, before starting hbase.
 prerequisite:
   - cluster-configuration
-  - hadoop-service
+  - hadoop-name-node
+  - hadoop-data-node
+  - zookeeper
 
 
 # paictl will generate the template file with the name "filename".template with jinja2.
```

Comment:
Line 159 has a badlink - Specify which label of roles you want to deploy your service in [services-configuration.yaml]

Reply:
Line 160, we can't show users the NodeAffinity, it's hide in code and can't be seen in yaml file, we just need to let users know we used it.